app/src/app/search/search.component.ts

```diff
@@ -133,23 +134,39 @@ export class SearchComponent implements OnInit, OnDestroy {
    * Triggers the actual search api call.
    */
   public onSearch(): void {
+    this.publicIdResult = undefined;
     this.spinner.show('app', 'Searching...');
     let params = new HttpParams();
     params = params.set('limit', this.limit.toString());
     params = params.set('skip', this.skip.toString());
     if (this.resolve.value) {
       params = params.set('resolve', this.resolve.value.toString());
     }
-    this.searchService.postSearch(this.queryParameters, params).subscribe((results: SearchResultList) => {
-      this.searchResultList = results;
-      this.filterResultList = this.initFilter ? results.groups : this.filterResultList;
-      if (this.initSearch) {
-        this.maxNumberOfSites = Array.from({ length: (this.searchResultList.total_results) }, (v, k) => k + 1);
-        this.initSearch = false;
-        this.initFilter = false;
-      }
-      this.spinner.hide('app');
+    this.searchService.postSearch(this.queryParameters, params).pipe(takeUntil(this.subscriber)).subscribe((results: SearchResultList) => {
+        this.searchResultList = results;
+        this.filterResultList = this.initFilter && results !== null ? results.groups : this.filterResultList;
+        if (this.initSearch) {
+          this.maxNumberOfSites = Array.from({length: (this.searchResultList.total_results)}, (v, k) => k + 1);
+          this.initSearch = false;
+          this.initFilter = false;
+        }
     });
+    let localParameters = JSON.parse(this.queryParameters);
```

Comment:
You could handle the spinner case with a
`.add(() => this.spinner.hide());`
At the end of the subscription. Little bit nicer, but optional.